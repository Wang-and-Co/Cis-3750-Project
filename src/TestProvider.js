/* eslint-disable react/destructuring-assignment */
import * as React from 'react';
import { createTheme } from '@mui/material';
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles';
// import defaultTheme from '../../src/renderer/data/theme';

export default class TestProvider extends React.Component {
  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children,
    },
  };

  render() {
    const theme = createTheme({
      //   ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions,
      },
    });
    return (
      // eslint-disable-next-line react/jsx-no-comment-textnodes
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
      </StyledEngineProvider>
    );
  }
}
