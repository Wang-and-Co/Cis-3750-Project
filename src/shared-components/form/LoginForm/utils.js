export const loginFormInitialValues = {
  email: '',
  password: '',
};
export const signUpFormInitialValues = {
  firstName: '',
  lastName: '',
  email: '',
  password: '',
  verifyPassword: '',
};
