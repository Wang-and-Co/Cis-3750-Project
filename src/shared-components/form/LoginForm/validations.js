import { object, string } from 'yup';

// getter for the validation schema. Is not a constant because we want an instance per usage
export const getLoginFormValidationSchema = () =>
  object({
    name: string().required('A value is required'),
    password: string()
      .required('A value is required')
      .min(8, 'Required 8 characters'),
  });
