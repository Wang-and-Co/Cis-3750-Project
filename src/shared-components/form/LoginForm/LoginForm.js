import { Form, Formik } from 'formik';
import { initialValues } from './utils';
import { getLoginFormValidationSchema } from './validations';
import { InputField } from '../InputField';
import { Button, Stack, Typography, Link } from '@mui/material';
/**
 * @typedef {Object} LoginFormProps
 * @property {function} onSuccess function handler after form is successfully submitted. Should handle the endpoint calls
 * with Backend for logging in.
 * @property {function} footerOnClick
 */

/**
 * A login form that contains everything required for logins, including the submit button.
 * @param {LoginFormProps} props - {@link LoginFormProps} object
 * @type {React.FC<LoginFormProps>}
 */
const LoginForm = ({ onSuccess, handleSubmit, footerOnClick }) => {
  const validationSchema = getLoginFormValidationSchema(); // get the validation schema from validations file
  return (
    <Formik
      initialValues={initialValues}
      validationSchema={validationSchema}
      validateOnChange={false}
      onSubmit={handleSubmit}
    >
      {(formikProps) => (
        <Form>
          <Stack spacing={1} sx={{ marginTop: '1.5rem' }}>
            <InputField
              name="email"
              label="Email"
              required
              autoComplete="email"
            ></InputField>
            <InputField
              name="password"
              label="Password"
              required
              type="password"
              autoComplete="current-password"
            ></InputField>
            <Button
              onClick={formikProps.handleSubmit}
              variant="contained"
              disabled={formikProps.isSubmitting}
            >
              Login
            </Button>
            <div style={{ display: 'flex' }}>
              <Typography inline variant={'subtitle2'}>
                First time here?&nbsp;
              </Typography>
              <Link
                inline
                variant={'subtitle2'}
                onClick={footerOnClick}
                disabled={formikProps.isSubmitting}
              >
                Sign up
              </Link>
            </div>
          </Stack>
        </Form>
      )}
    </Formik>
  );
};
export default LoginForm;
