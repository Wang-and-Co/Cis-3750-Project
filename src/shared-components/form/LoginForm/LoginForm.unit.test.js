import { create } from 'react-test-renderer';
import TestProvider from '../../../TestProvider';
import LoginForm from './LoginForm';

jest.mock('./utils', () => ({
  _esModule: true,
  initialValues: {
    name: 'test',
    password: 'this',
  },
}));

const props = {
  onSuccess: jest.fn(),
};

let loginFormRenderer;
let loginFormInstance;

beforeEach(async () => {
  await renderLoginForm(props);
});

afterEach(() => {
  loginFormRenderer.unmount();
});
describe('Generic Modal', () => {
  it('renders correctly', () => {
    expect(loginFormRenderer.toJSON()).toMatchSnapshot();
  });
  it('passes the correct props', () => {
    expect(GetFormProp('onSuccess')).toBe(props.onSuccess);
  });
});

const GetFormProp = (prop) =>
  loginFormInstance.findByType(LoginForm).props[prop];

const renderLoginForm = async (editorProps) => {
  loginFormRenderer = await create(
    <TestProvider>
      <LoginForm {...editorProps}></LoginForm>
    </TestProvider>,
  );
  loginFormInstance = loginFormRenderer.root;
};
