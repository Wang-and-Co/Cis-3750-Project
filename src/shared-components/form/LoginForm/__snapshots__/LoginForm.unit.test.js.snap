// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic Modal renders correctly 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="MuiStack-root css-hxrey1-MuiStack-root"
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-1u6ro6r-MuiFormControl-root-MuiTextField-root"
      onKeyDown={[Function]}
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={true}
        htmlFor="input-email"
        id="input-email-label"
      >
        Email
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="email"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="input-email"
          name="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value="test@test.com"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-14lo706"
          >
            <span>
              Email
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root css-1u6ro6r-MuiFormControl-root-MuiTextField-root"
      onKeyDown={[Function]}
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={true}
        htmlFor="input-password"
        id="input-password-label"
      >
        Password
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="current-password"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="input-password"
          name="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value="this"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-14lo706"
          >
            <span>
              Password
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Login
    </button>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
        inline={true}
      >
        First time here? 
      </h6>
      <a
        className="MuiTypography-root MuiTypography-subtitle2 MuiLink-root MuiLink-underlineAlways css-66npy3-MuiTypography-root-MuiLink-root"
        inline={true}
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Sign up
      </a>
    </div>
  </div>
</form>
`;
